### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ThePrivatePanda**
This PR brings alot of chamge.
- Use `isort` and `black`
- Clean up code, reptitive functions, unused imports/functions
- Fixed some improper handling of ws
- General code improvement

### CODE REVIEW

File: src/DankCord/DankCord.py [hunk 1]
```diff
@@ -1,260 +1,297 @@
...
-import json, datetime, time
-
-from rich import print, print_json
-from typing import Optional, Union
-from .Objects import Response, Message
-from .exceptions import InvalidToken, UnknownChannel
-from .ws import Gateway
+import datetime
+import json
+import time
```
Comment (theAbdoSabbagh) [inline src/DankCord/DankCord.py:3]: I'm not sure if this would affect performance or anything but wouldn't it be better to put those imports in 1 line instead?

File: src/DankCord/DankCord.py [hunk 2]
```diff
@@ -1,260 +1,297 @@
...
 from string import printable
 from threading import Thread
+from typing import Optional, Literal
+
+import faster_than_requests as requests
+import orjson
```
Comment (theAbdoSabbagh) [inline src/DankCord/DankCord.py:9]: Same here

File: src/DankCord/DankCord.py [hunk 3]
```diff
@@ -1,260 +1,297 @@
...
+        data = {
+            "type": 2,
+            "application_id": "270904126974590976",
+            "guild_id": self.guild_id,
+            "channel_id": self.channel_id,
+            "session_id": self.session_id,
+            "data": {
                 "version": command_info["version"],
-                "default_permission": command_info["default_permission"],
-                "default_member_permissions": command_info["default_member_permissions"],
```
Comment (theAbdoSabbagh) [inline src/DankCord/DankCord.py:139]: Why did you remove those? In the normal Discord clients, requests have their payload with those pieces of data, so I replicated them

File: src/DankCord/DankCord.py [hunk 4]
```diff
@@ -1,260 +1,297 @@
...
+            "application_id": "270904126974590976",
+            "guild_id": self.guild_id,
+            "channel_id": self.channel_id,
+            "session_id": self.session_id,
+            "data": {
+                "version": command_info["version"],
+                "id": command_info["id"],
                 "name": name,
-                "description": command_info["description"],
-                "dm_permission": command_info["dm_permission"],
```
Comment (theAbdoSabbagh) [inline src/DankCord/DankCord.py:143]: Same here

File: src/DankCord/gateway.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        self.logger.bootup("Discord heartbeat client.")
+
+        jitter_heartbeat = self.heartbeat_interval * random.uniform(0, 0.1)
+
+        time.sleep(jitter_heartbeat)
+        ws.send(orjson.dumps({"op": 1, "d": None}))
+        response = orjson.loads(ws.recv())
+
+        if response["op"] != 11:
+            print("Discord heartbeat client failed to boot.")
```
Comment (theAbdoSabbagh) [inline src/DankCord/gateway.py]: I suggest changing this line to
```py
self.logger.error("Discord heartbeat client failed to boot.")
```
Or, perhaps raising an error instead? Whichever is better

File: src/DankCord/DankCord.py (hunk 1 above)
Comment (ThePrivatePanda) [reply src/DankCord/DankCord.py:3]: PEP8

File: src/DankCord/DankCord.py (hunk 2 above)
Comment (ThePrivatePanda) [reply src/DankCord/DankCord.py:9]: PEP8

File: src/DankCord/gateway.py (hunk 5 above)
Comment (ThePrivatePanda) [reply src/DankCord/gateway.py]: Implemented in next commit

File: src/DankCord/DankCord.py (hunk 3 above)
Comment (ThePrivatePanda) [reply src/DankCord/DankCord.py:139]: Hasn't been removed, it's still there. Probably a bug with the UI or something?

File: src/DankCord/DankCord.py (hunk 4 above)
Comment (ThePrivatePanda) [reply src/DankCord/DankCord.py:143]: Same as above

**Review by theAbdoSabbagh — APPROVED** [review]
